providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/process/storage/connector/grpc/GrpcStreamConnector.java

```diff
@@ -0,0 +1,137 @@
+import dev.openfeature.contrib.providers.flagd.resolver.process.storage.connector.StreamPayload;
+import dev.openfeature.contrib.providers.flagd.resolver.process.storage.connector.StreamPayloadType;
+import dev.openfeature.flagd.sync.FlagSyncServiceGrpc;
+import dev.openfeature.flagd.sync.SyncService;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import io.grpc.ManagedChannel;
+import lombok.extern.java.Log;
+
+import java.util.Random;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.logging.Level;
+
+/**
+ * Implements the {@link Connector} contract and emit flags obtained from flagd sync gRPC contract.
+ */
+@Log
+@SuppressFBWarnings(value = {"PREDICTABLE_RANDOM", "EI_EXPOSE_REP"},
+        justification = "Random is used to generate a variation & flag configurations require exposing")
+public class GrpcStreamConnector implements Connector {
+    private static final Random RANDOM = new Random();
+
+    private static final int INIT_BACK_OFF = 2 * 1000;
+    private static final int MAX_BACK_OFF = 120 * 1000;
+
+    private final AtomicBoolean shutdown = new AtomicBoolean(false);
+    private final BlockingQueue<StreamPayload> blockingQueue = new LinkedBlockingQueue<>(5);
+
+    private final ManagedChannel channel;
+    private final FlagSyncServiceGrpc.FlagSyncServiceStub serviceStub;
+
+    public GrpcStreamConnector(final FlagdOptions options) {
+        channel = ChannelBuilder.nettyChannel(options);
+        serviceStub = FlagSyncServiceGrpc.newStub(channel);
+    }
+
+    /**
+     * Initialize gRPC stream connector.
+     */
+    public void init() {
+        Thread listener = new Thread(() -> {
+            try {
+                observeEventStream(blockingQueue, shutdown, serviceStub);
+            } catch (InterruptedException e) {
+                log.log(Level.WARNING, "gRPC event stream interrupted, flag configurations are stale", e);
+            }
+        });
+
+        listener.setDaemon(true);
+        listener.start();
+    }
+
+    /**
+     * Get blocking queue to obtain payloads exposed by this connector.
+     */
+    public BlockingQueue<StreamPayload> getStream() {
+        return blockingQueue;
+    }
+
+    /**
+     * Shutdown gRPC stream connector.
+     */
+    public void shutdown() {
+        if (shutdown.getAndSet(true)) {
+            return;
+        }
+
+        channel.shutdown();
+    }
+
+    /**
+     * Contains blocking calls, to be used concurrently.
+     */
+    static void observeEventStream(final BlockingQueue<StreamPayload> writeTo,
+                                   final AtomicBoolean shutdown,
+                                   final FlagSyncServiceGrpc.FlagSyncServiceStub serviceStub)
+            throws InterruptedException {
+
+        final BlockingQueue<GrpcResponseModel> blockingQueue = new LinkedBlockingQueue<>(5);
```

Comment:
we use number 5 on multiple places, is it worth to introduce a constant and use that instead ?

Reply:
done :)